rules-java-diva/addon/src/main/java/org/jboss/windup/rules/apps/diva/analysis/DivaLauncher.java

```diff
@@ -0,0 +1,613 @@
+                                            cs.add(GraphService.addTypeToModel(gc, model,
+                                                    DivaRequestConstraintModel.class));
+                                        }
+                                    }
+                                });
+                                map.put(Report.TRANSACTIONS, txs);
+                            });
+                        }
+                    }, txAnalysis);
+                }
+                gc.getGraph().tx().commit();
+                success++;
+            } catch (RuntimeException e) {
+                gc.getGraph().tx().rollback();
+                failure++;
+            }
+            if ((success + failure) % 10 == 0) {
+                LOG.info("Diva: transaction analysis: " + (success + failure) + "/" + contexts.size() + " (" + failure
+                        + " failures)");
+            }
+        }
+
+        endpointResolution(gc, projects);
+
+        LOG.info("Diva: DONE");
+    }
+
+    public static List<MavenProjectModel> getMavenDeps(MavenProjectModel m, List<MavenProjectModel> knownProjects) {
+        List<MavenProjectModel> candidates = Util
+                .makeList(Util.map(Util.filter(Util.map(m.getDependencies(), ProjectDependencyModel::getProjectModel),
+                        d -> (d instanceof MavenProjectModel)), p -> (MavenProjectModel) p));
+        List<MavenProjectModel> res = new ArrayList<>();
+        for (MavenProjectModel known : knownProjects) {
+            if (Util.any(candidates, c -> c.getGroupId().equals(known.getGroupId())
+                    && c.getArtifactId().equals(known.getArtifactId()))) {
+                res.add(known);
+            }
+        }
+        return res;
+    }
+
+    public static String stripBraces(String s) {
+        StringBuilder b = new StringBuilder();
+        for (int i = 0; i < s.length(); i++) {
+            b.append(s.charAt(i));
+            if (s.charAt(i) == '{') {
+                for (; s.charAt(i) != '}'; i++)
+                    ;
+                b.append('}');
+            }
+        }
+        return b.toString();
+    }
+
+    public static void endpointResolution(GraphContext gc, List<? extends ProjectModel> projects) {
+
+        // 1) obtaining project -> application.properties mapping
+
+        Map<String, Properties> appProps = new LinkedHashMap<>();
+        FileService files = new FileService(gc);
+        for (ProjectModel p : projects) {
+            String targetPath = Paths.get(p.getRootFileModel().getFilePath())
+                    .resolve("src/main/resources/application.properties").toString();
+            PropertiesModel file = (PropertiesModel) files.findByPath(targetPath);
+            if (file != null) {
+                try {
+                    appProps.put(p.getName(), file.getProperties());
+                } catch (IOException e) {
```

Comment:
Perhaps log an error here so it doesn't just get swallowed and ignored?

Reply:
See c1ee067